singer_sdk/tap_base.py

```diff
@@ -338,6 +338,7 @@ def _singer_catalog(self) -> Catalog:
         return Catalog(
             (stream.tap_stream_id, stream._singer_catalog_entry)  # noqa: SLF001
             for stream in self.streams.values()
+            if not stream.__abstract__
```

Comment:
Advantages/disadvantages of filtering out abstract streams here vs in `self.streams`?